subquery/src/mappings/ens.ts

```diff
@@ -0,0 +1,131 @@
+import {
+  RegisterTransaction,
+  TransferLog,
+} from "./../types/abi-interfaces/ENSAbi";
+import { fetchAccount, fetchTransaction } from "../utils/utils";
+import { ENS } from "../types";
+import { fetchContract } from "../utils/erc721";
+import { ethers } from "ethers";
+
+function generateId(name: string) {
+  const hash = ethers.utils.keccak256(ethers.utils.toUtf8Bytes(name));
+  const uint256 = ethers.BigNumber.from(hash).toString();
+  return uint256;
+}
+
+export async function handleCallRegistry(tx: RegisterTransaction) {
+  const receipt = await tx.receipt();
+
+  if (!receipt.status) {
+    return;
+  }
+
+  if (!tx.args || !tx.logs) {
+    throw new Error("No tx.args or tx.logs");
+  }
+
+  const [owner, name] = tx.args;
+  const txHash = tx.hash;
+  const idx = tx.logs.findIndex((log) => {
+    return (
+      log.transactionHash === txHash &&
+      !!log.args &&
+      log.args[0] !== "0x0000000000000000000000000000000000000000" &&
+      log.args[1] === owner
+    );
+  });
+
+  const timestamp = BigInt(tx.blockTimestamp) * BigInt(1000);
+  const ownerAccount = await fetchAccount(owner, timestamp);
+  const caller = await fetchAccount(String(tx.from).toLowerCase(), timestamp);
+  const txIndex = tx.transactionIndex;
+
+  const event = tx.logs[idx];
+
+  const [rawOwner, _, expires] = event.args!;
+
+  const expiresAt = expires ? expires.toBigInt() * BigInt(1000) : BigInt(0);
+  // rawName is the encoded name
+  const rawName = JSON.stringify(rawOwner);
```

Comment:
I'm not sure if there is any value in reading and storing the encoded name. It should be simply a hex convert of the string isn't it?